lib/analyzer/applications/python.ts

```diff
@@ -0,0 +1,90 @@
+import * as path from "path";
+import * as lockFileParser from "snyk-poetry-lockfile-parser";
+import { DepGraphFact, TestedFilesFact } from "../../facts";
+
+import { DepGraph } from "@snyk/dep-graph";
+import { AppDepsScanResultWithoutTarget, FilePathToContent } from "./types";
+
+interface ManifestLockPathPair {
+  manifest: string;
+  lock: string;
+}
+
+export async function poetryFilesToScannedProjects(
+  filePathToContent: FilePathToContent,
+): Promise<AppDepsScanResultWithoutTarget[]> {
+  const scanResults: AppDepsScanResultWithoutTarget[] = [];
+
+  const filePairs = findManifestLockPairsInSameDirectory(filePathToContent);
+
+  const shouldIncludeDevDependencies = false;
+
+  for (const pathPair of filePairs) {
+    const depGraph = await lockFileParser.buildDepGraph(
+      filePathToContent[pathPair.manifest],
+      filePathToContent[pathPair.lock],
+      shouldIncludeDevDependencies,
+    );
+
+    const depGraphFact: DepGraphFact = {
+      type: "depGraph",
+      data: depGraph as DepGraph,
+    };
+    const testedFilesFact: TestedFilesFact = {
+      type: "testedFiles",
+      data: [path.basename(pathPair.manifest), path.basename(pathPair.lock)],
+    };
+    scanResults.push({
+      facts: [depGraphFact, testedFilesFact],
+      identity: {
+        type: depGraph.pkgManager.name,
+        targetFile: pathPair.manifest,
+      },
+    });
+  }
+
+  return scanResults;
+}
+
+function findManifestLockPairsInSameDirectory(
+  filePathToContent: FilePathToContent,
+): ManifestLockPathPair[] {
+  const fileNamesGroupedByDirectory = groupFilesByDirectory(filePathToContent);
+  const manifestLockPathPairs: ManifestLockPathPair[] = [];
+
+  for (const directoryPath of Object.keys(fileNamesGroupedByDirectory)) {
```

Comment:
Not related to this PR, but a general TS question: is there no "nice way" of looping over the keys & values of a map? 
Something like Go's `for k, v := range m { ... }`.

Reply:
There's nothing similar in JavaScript. The closest thing are [for..in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) loops, but they are generally not recommended